engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -953,14 +949,21 @@ class Compiler(
   private def reportDiagnostics(
     diagnostics: List[Diagnostic],
     compilerModule: CompilerContext.Module
-  ): Boolean = {
+  ): List[RuntimeException] = {
     val isOutputRedirected = config.outputRedirect.isDefined
-    diagnostics.foreach { diag =>
-      val formattedDiag =
-        context.formatDiagnostic(compilerModule, diag, isOutputRedirected)
-      printDiagnostic(formattedDiag)
-    }
-    diagnostics.exists(_.isInstanceOf[Error])
+    val exceptions = diagnostics
+      .map { diag =>
+        val formattedDiag =
+          context.formatDiagnostic(compilerModule, diag, isOutputRedirected)
+        printDiagnostic(formattedDiag.getMessage)
+        if (diag.isInstanceOf[Error]) {
+          formattedDiag
+        } else {
+          null
+        }
+      }
+      .filter(_ != null)
```

Comment:
`flatMap` on `Option` would be more Scala'ish

Reply:
Yes, it is nicer: cda1a598e9 (if it does the same).